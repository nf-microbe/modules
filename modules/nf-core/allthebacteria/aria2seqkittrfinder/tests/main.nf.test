nextflow_process {

    name "Test Process ALLTHEBACTERIA_ARIA2SEQKITTRFINDER"
    script "../main.nf"
    process "ALLTHEBACTERIA_ARIA2SEQKITTRFINDER"

    tag "modules"
    tag "modules_nfcore"
    tag "allthebacteria"
    tag "allthebacteria/aria2seqkittrfinder"

    test("atb.assembly.incr_release.202408.batch.1.tar.xz") {
        when {
            process {
                """
                input[0] = [
                    [ id:'atb.assembly.incr_release.202408.batch.1.tar.xz' ], // meta map
                    [
                        "https://osf.io/download/bfeqh/"
                    ]
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("atb.assembly.incr_release.202408.batch.1.tar.xz - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'achromobacter_xylosoxidans__01' ], // meta map
                    [
                        "https://osf.io/download/bfeqh/"
                    ]
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
