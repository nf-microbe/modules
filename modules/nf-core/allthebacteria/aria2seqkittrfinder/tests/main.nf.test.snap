{
    "atb.assembly.incr_release.202408.batch.1.tar.xz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "atb.assembly.incr_release.202408.batch.1.tar.xz"
                        },
                        "atb.assembly.incr_release.202408.batch.1.tar.xz.trfinder.fasta.gz:md5,adbf4eabc203c99ab102737edb8b3bba"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "atb.assembly.incr_release.202408.batch.1.tar.xz"
                        },
                        "atb.assembly.incr_release.202408.batch.1.tar.xz.trfinder.tsv:md5,e1a7850e590af17a2337ac882f0a4a76"
                    ]
                ],
                "2": [
                    "versions.yml:md5,4d537387f97a1af2fac93bb00835f3e4"
                ],
                "fasta": [
                    [
                        {
                            "id": "atb.assembly.incr_release.202408.batch.1.tar.xz"
                        },
                        "atb.assembly.incr_release.202408.batch.1.tar.xz.trfinder.fasta.gz:md5,adbf4eabc203c99ab102737edb8b3bba"
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "atb.assembly.incr_release.202408.batch.1.tar.xz"
                        },
                        "atb.assembly.incr_release.202408.batch.1.tar.xz.trfinder.tsv:md5,e1a7850e590af17a2337ac882f0a4a76"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4d537387f97a1af2fac93bb00835f3e4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-21T09:15:34.98738437"
    },
    "atb.assembly.incr_release.202408.batch.1.tar.xz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "achromobacter_xylosoxidans__01"
                        },
                        "achromobacter_xylosoxidans__01.trfinder.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "achromobacter_xylosoxidans__01"
                        },
                        "achromobacter_xylosoxidans__01.trfinder.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,4d537387f97a1af2fac93bb00835f3e4"
                ],
                "fasta": [
                    [
                        {
                            "id": "achromobacter_xylosoxidans__01"
                        },
                        "achromobacter_xylosoxidans__01.trfinder.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "achromobacter_xylosoxidans__01"
                        },
                        "achromobacter_xylosoxidans__01.trfinder.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4d537387f97a1af2fac93bb00835f3e4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-21T09:16:19.540946236"
    }
}